runtime/permissions.rs

```diff
@@ -83,10 +111,11 @@ impl UnitPermission {
 
   pub fn request(&mut self) -> PermissionState {
     if self.state == PermissionState::Prompt {
-      if permission_prompt(&format!("access to {}", self.description)) {
-        self.state = PermissionState::Granted;
-      } else {
-        self.state = PermissionState::Denied;
+      match permission_prompt(&format!("access to {}", self.description)) {
+        PromptResult::AllowAlways => self.state = PermissionState::Granted,
+        PromptResult::AllowOnce => return PermissionState::Granted,
+        PromptResult::DenyOnce => return PermissionState::Denied,
```

Comment:
What do allow once and deny once mean for `Deno.permisssions.request()`? This should still use the current prompt.